helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -0,0 +1,120 @@
+package org.apache.helix.model;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.HelixProperty;
+import org.apache.helix.ZNRecord;
+
+/**
+ * Cloud configurations
+ */
+public class CloudConfig extends HelixProperty {
+  /**
+   * Configurable characteristics of a cloud.
+   * NOTE: Do NOT use this field name directly, use its corresponding getter/setter in the
+   * CloudConfig.
+   */
+  private enum CloudConfigProperty {
+    CLOUD_ENABLED, // determine whether the cluster is inside cloud environment.
+    CLOUD_ID, // the cloud Id that belongs to this cluster.
+    CLOUD_INFO_FETCHER, // the url from where to retrieve the cloud information.
+    CLOUD_INFO_PARSER, // the name of the function that parses the topology.
+  }
+
+  /**
+   * Instantiate the CloudConfig for the cloud
+   * @param cluster
+   */
+  public CloudConfig(String cluster) {
```

Comment:
We can think about the builder here, since we need to verify the completion of cloud config. For example, if you enable cloud config but there is no URL or parser class. Shall we fail it?

For details, you can refer WorkflowConfig.

Reply:
Updated.